src/Akka.Logger.NLog/NLogLogger.cs

```diff
@@ -26,51 +26,52 @@ public class NLogLogger : ReceiveActor, IRequiresMessageQueue<ILoggerMessageQueu
     {
         private readonly ILoggingAdapter _log = Logging.GetLogger(Context.System.EventStream, "NLogLogger");
 
-        private static void Log(LogEvent logEvent, Action<NLogger, LogEvent> logStatement)
-        {
-            var loggerName = (logEvent.LogClass == typeof(DummyClassForStringSources) || logEvent.LogClass.GenericTypeArguments?.Length != 0)
-                ? logEvent.LogSource
-                : logEvent.LogClass.ToString(); // Include full namespace, but not assembly name
-            var logger = LogManager.GetLogger(loggerName);
-            logStatement(logger, logEvent);
-        }
-
         /// <summary>
         /// Initializes a new instance of the <see cref="NLogLogger"/> class.
         /// </summary>
         public NLogLogger()
         {
-            Receive<Error>(m => Log(m, (logger, logEvent) => LogEvent(logger, NLogLevel.Error, logEvent.Cause, logEvent)));
-            Receive<Warning>(m => Log(m, (logger, logEvent) => LogEvent(logger, NLogLevel.Warn, logEvent.Cause, logEvent)));
-            Receive<Info>(m => Log(m, (logger, logEvent) => LogEvent(logger, NLogLevel.Info, logEvent.Cause, logEvent)));
-            Receive<Debug>(m => Log(m, (logger, logEvent) => LogEvent(logger, NLogLevel.Debug, logEvent.Cause, logEvent)));
+            Receive<Error>(static evt => LogEvent(evt, NLogLevel.Error));
```

Comment:
that's very clever - is this to prevent delegate allocations @snakefoot ?

Reply:
Yes the `static`-keyword prevents one from accidental introducing delegate capture in future refactoring.

Reply:
Well, I think we probably have a ton of areas where we could use that in the main Akka.NET project. I've used static local functions for some of this before but doing it on an inline lambda is a new one for me